### ISSUE TITLE
Improve qube indication in the notifications proxy

### ISSUE THREAD

**Issue by marmarek** [QubesOS/qubes-issues#9648]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

The qubes-notification-proxy currently just prepends the qube name to the notification text. It's not very well visible. Old notification approach (notification daemon in each qube) had small colorful border around each notification - while not having full qube name, including the color was IMO a good idea to quickly distinguish notifications from different qubes.

### The solution you'd like

Include qube color in the notification somewhere. The easiest thing to do is probably to simply use qube's icon as the notification icon. And if the notification has an icon already, merge those two (add qube's own icon as an overlay in the right bottom corner, over the original notification icon).
And maybe also make the qube name more distinct (in bold? or something?).

### The value to a user, and who that user might be

All users, since the qubes-notification-proxy is considered to be default in R4.3.

### Completion criteria checklist

(This section is for developer use only. Please do not modify it.)

**Comment by marmarek** [comment]
@marmarta we have discussed this. Today, but also some months ago somewhere (Matrix?). Feel free to comment if I missed anything.

**Comment by marmarta** [comment]
The notification should also get the colored border (please don't tell me that "ooh, we can't actually do it easily over wayland", I don't care, if wayland can't do anything we shouldn't be using it)

**Comment by UndeadDevel** [comment]
An idea: what if qube notifications would display with a button (i.e. `notify-send` `-A` option), like "suppress notifications from this qube"? E.g. `notify-send -A Yes -A No -A "Suppress notifications from this qube" "Execute task?" "Execute your scheduled task xyz now?"`, which shows that two additional buttons "from the qube" can be displayed also; the proxy would then relay the answer except if the third button is pressed, in which case it will stop serving notifications from that qube.
That would make it even more obvious, even without a border, that the notification isn't from dom0, but I haven't looked at how the proxy works, so maybe that's not possible, either.

**Comment by UndeadDevel** [comment]
> There is one small issue with the additional button: it won't be that obvious where to put the option to enable those back (hard to discover)

I think intuitively users would look either in Qubes Global Config (if a "Notifications" tab were added there) or in the regular per-qube settings.

> we'd also need to test how it looks with multiple buttons if a notification by itself gives you two or three buttons.

Extreme case: 
![notification](https://github.com/user-attachments/assets/4197b73a-5b50-4da4-b48c-12c5ea7f95a5)

**Comment by marmarta** [comment]
That looks good (and good point on global config)

**Comment by UndeadDevel** [comment]
I'll add a less extreme case with three qube buttons and the full desktop in view:

![notification](https://github.com/user-attachments/assets/ba9bc0a6-7d82-4f77-b0dd-957888e93c8b)

So it does inflate the size of notifications with buttons noticeably, but only if there's buttons in the notification from the qube's side, which normally isn't the case:

![notification](https://github.com/user-attachments/assets/6e253556-3a82-4237-9383-1fcdd7055c54)

**Comment by marmarek** [comment]
NetworkManager notifications already have "don show again" button, see https://openqa.qubes-os.org/tests/121782#step/firstboot/72. It will be quite confusing to have two such buttons on a single notification...

**Comment by marmarta** [comment]
I don't agree - there is a difference between "Suppress future notifications from this qube" (subject to rewording by a less-wordy native speaker) and "Don't show this notification in the future"

**Comment by UndeadDevel** [comment]
> What if VM will really try to put a button "Suppress future notifications from this qube" too? (possibly with intentional typo, to bypass potential check for exact same wording)

I suppose the user would know that the right-most button is the one always added, unless the user hasn't used that Qubes version for long, but in any case: the button is supposed to enable the user to stop a kind of DOS attack on the notification system, right? So even if the user clicks the wrong button first, the next time the user will likely try the other one.

For reference (didn't find the proper icon):

![notification](https://github.com/user-attachments/assets/b32a177b-b1c3-414b-8cda-bf75f8491215)


![notification](https://github.com/user-attachments/assets/bfba8786-7853-4caa-ab00-3086dc4f9f99)

**Comment by DemiMarie** [comment]
> The notification should also get the colored border (please don't tell me that "ooh, we can't actually do it easily over wayland", I don't care, if wayland can't do anything we shouldn't be using it).

This is actually doable, _if_ we patch the notification daemon.  I expect that the patch will not be upstreamable.

**Comment by ben-grande** [comment]
> > The notification should also get the colored border (please don't tell me that "ooh, we can't actually do it easily over wayland", I don't care, if wayland can't do anything we shouldn't be using it).
> 
> This is actually doable, _if_ we patch the notification daemon. I expect that the patch will not be upstreamable.

I don't think we can guarantee which notification daemon is used. 

---

> Include qube color in the notification somewhere. The easiest thing to do is probably to simply use qube's icon as the notification icon. And if the notification has an icon already, merge those two (add qube's own icon as an overlay in the right bottom corner, over the original notification icon).
> And maybe also make the qube name more distinct (in bold? or something?).

I read the whole thread, what remains be done here?

Is this?

> And if the notification has an icon already, merge those two (add qube's own icon as an overlay in the right bottom corner, over the original notification icon).

**Comment by marmarek** [comment]
> Is this?
> 
> > And if the notification has an icon already, merge those two (add qube's own icon as an overlay in the right bottom corner, over the original notification icon).

Yes.

**Comment by marmarta** [comment]
<img width="1058" height="282" alt="Image" src="https://github.com/user-attachments/assets/7f57bd27-7a36-46e0-ae98-f422293202cd" />

**Comment by ben-grande** [comment]
What is "signal" qube? The last qube you connected this device?

**Comment by marmarta** [comment]
This is an example of how the icon should look like, from discussion we had way back when in the matrix channel; I don't recall what was the exact idea being discussed there. The point is the icon :)

**Comment by ben-grande** [comment]
- https://github.com/QubesOS/qubes-notification-proxy/pull/19
